SampleGetApi/ReportsHelper/ReportUtils.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Xml.Linq;
+using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+
+namespace SampleGetApi.ReportsHelper;
+
+	public class ReportUtils
+	{
+    public static ExtentReports? ExtentReport;
+
+    public static void Initalize()
+		{
+        var reportFolder = CreateReportFolder();
+
+        if (reportFolder != null || reportFolder != "")
+        {
+            var indexPath = Path.Combine(reportFolder, "index.html");
+
+            var spark = new ExtentSparkReporter(indexPath);
+            ExtentReport = new ExtentReports();
+            ExtentReport.AttachReporter(spark);
+            //Extent.AddSystemInfo("Host Name", System.Net.Dns.GetHostName());
+            //Extent.AddSystemInfo("Environment", "QA");
+            //Extent.AddSystemInfo("User Name", Environment.UserName);
+        }
+        else
+        {
+            Console.WriteLine("Error creating report folder.");
+        }
+    }
+
+    public static void CloseReport()
+		{
+        if (ExtentReport != null)
+        {
+            ExtentReport.Flush();
+        }
+        else
+        {
+            Console.WriteLine("Extent object is null. Make sure to call Initalize method first.");
+        }
+    }
+
+    private static string CreateReportFolder()

```

Comment:
Are we not able to do it via config?

Reply:
config will give one single folder path and that has to change based on different machine/environment we are running. Plus we are creating different folders based on timestamp to keep a stack of executed reports.